Assets/BossRoom/Scripts/Shared/Infrastructure/UpdateRunner.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace BossRoom.Scripts.Shared.Infrastructure
+{
+
+    /// <summary>
+    /// Some objects might need to be on a slower update loop than the usual MonoBehaviour Update and without precise timing, e.g. to refresh data from services.
+    /// Some might also not want to be coupled to a Unity object at all but still need an update loop.
+    /// </summary>
+    public class UpdateRunner : MonoBehaviour
+    {
+        private class Subscriber
+        {
+            public Action<float> updateMethod;
+            public readonly float period;
+            public float periodCurrent;
+            public Subscriber(Action<float> updateMethod, float period)
+            {
+                this.updateMethod = updateMethod;
+                this.period = period;
+                this.periodCurrent = 0;
+            }
+        }
+
+        private List<Subscriber> m_subscribers = new List<Subscriber>();
+
+        public void OnDestroy()
+        {
+            m_subscribers.Clear(); // We should clean up references in case they would prevent garbage collection.
+        }
+
+        /// <summary>
+        /// Subscribe in order to have onUpdate called approximately every period seconds (or every frame, if period <= 0).
+        /// Don't assume that onUpdate will be called in any particular order compared to other subscribers.
+        /// </summary>
+        public void Subscribe(Action<float> onUpdate, float period)
+        {
+            if (onUpdate == null)
+            {
+                return;
+            }
+
+            if (onUpdate.Target == null) // Detect a local function that cannot be Unsubscribed since it could go out of scope.
+            {
+                Debug.LogError("Can't subscribe to a local function that can go out of scope and can't be unsubscribed from");
+                return;
+            }
+
+            if (onUpdate.Method.ToString().Contains("<")) // Detect
+            {
+                Debug.LogError("Can't subscribe with an anonymous function that cannot be Unsubscribed, by checking for a character that can't exist in a declared method name.");
+                return;
+            }
+
+            foreach (Subscriber currSub in m_subscribers)
+                if (currSub.updateMethod.Equals(onUpdate))
+                    return;
+            m_subscribers.Add(new Subscriber(onUpdate, period));
+        }
+        /// <summary>Safe to call even if onUpdate was not previously Subscribed.</summary>
+        public void Unsubscribe(Action<float> onUpdate)
+        {
+            for (int sub = m_subscribers.Count - 1; sub >= 0; sub--)

```

Comment:
I see these brace less blocks of code a bit everywhere, let's try to keep braces everywhere (brace less code is the source of a lot of evil and some nasty bugs) (coding convention says braces are required for loops, optional for ifs --> this https://blog.codecentric.de/en/2014/02/curly-braces/ is the reason I was traumatized by the lack of braces)

Reply:
(and yes I know the article is saying there's other ways to avoid this particular bug, but still :P )

Reply:
Damn it. I knew I shouldn't have trusted that Code Cleanup... I'll drop a formatting bomb on all of this.